xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxCC.cpp

```diff
@@ -12,9 +12,107 @@
 #include "cores/VideoPlayer/DVDCodecs/Overlay/contrib/cc_decoder708.h"
 #include "cores/VideoPlayer/Interface/CaptionBlock.h"
 #include "cores/VideoPlayer/Interface/TimingConstants.h"
+#include "utils/StringUtils.h"
 
 #include <algorithm>
 
+namespace
+{
+/*!
+ * \brief Given the color code from the close caption decoder, returns the corresponding
+ * color in rgb format (striping the alpha channel from argb)
+ * 
+ * \param[in] ccColor - a given color from the CTA608 decoder
+ * \return the corresponding Color in rgb
+ */
+UTILS::COLOR::Color Cea608ColorConversion(const uint8_t ccColor)
+{
+  UTILS::COLOR::Color color = UTILS::COLOR::NONE;
+  switch (ccColor)
+  {
+    case WHITE:
+      color = UTILS::COLOR::WHITE;
+      break;
+    case GREEN:
+      color = UTILS::COLOR::GREEN;
+      break;
+    case BLUE:
+      color = UTILS::COLOR::BLUE;
+      break;
+    case CYAN:
+      color = UTILS::COLOR::CYAN;
+      break;
+    case RED:
+      color = UTILS::COLOR::RED;
+      break;
+    case YELLOW:
+      color = UTILS::COLOR::YELLOW;
+      break;
+    case MAGENTA:
+      color = UTILS::COLOR::MAGENTA;
+      break;
+    case BLACK:
+      color = UTILS::COLOR::BLACK;
+      break;
+    default:
+      break;
+  }
+  return color & ~0xFF000000;
+}
+
+/*!
+ * \brief Given the current buffer cc text and the decoder row
+ * it computes the text to account for the presence of modifiers
+ * (italics, underline and text color)
+ * \param[in,out] ccText - the text to display in the caption
+ * \param[in] ccRow - The corresponding raw (including the current position and row cells)
+ */
+void ApplyStyleModifiers(std::string& ccText, const cc_row_t& ccRow)
+{
+  CaptionStyleProps style;
+  const int cursorPosition = ccRow.pos;
+  // ANSI CTA-608E:
+  // If, when a displayable character is received, it overwrites an existing PAC
+  // or mid-row code, and there are already characters to the right of the new character,
+  // these existing characters shall assume the same attributes as the new character.
+  // This adoption can result in a whole caption row suddenly changing color,
+  // underline, italics, and/or flash attributes.
+  for (int pos = cursorPosition; pos >= 0 && pos <= CC_COLUMNS; pos--)
+  {
+    const cc_attribute_t ccAttribute = ccRow.cells[pos].attributes;
+    if (ccAttribute.italic > 0)
+    {
+      // italics resets color to white
+      style.m_fontColor = UTILS::COLOR::WHITE;
+      style.m_italic = true;
+    }
+    if (ccAttribute.underline > 0)
+    {
+      style.m_underline = true;
+    }
+    if (ccAttribute.foreground > WHITE && ccAttribute.foreground <= BLACK && !style.m_italic)
+    {
+      style.m_fontColor = Cea608ColorConversion(ccAttribute.foreground);
+    }
+  }
+
+  // Apply style modifiers to CC text
+  if (style.m_italic)
+  {
+    ccText = StringUtils::Format("<i>{}</i>", ccText);
```

Comment:
in a separate PR i think is possible drop the use of sami converter (CDVDSubtitleTagSami) in the DVDOverlayCodecCCText.cpp so save a bit of computation and pass directly libass tags instead of SRT tags

Reply:
I would prefer to have tags that are not bound to a specific type (e.g. ass). So that the codec can use a tagger to convert them to the output format. 
I feel using the one we have already is the best option we have at the moment.